components/fxa-client/src/push.rs

```diff
@@ -35,6 +35,7 @@ impl FirefoxAccount {
                 Ok(vec![AccountEvent::ProfileUpdated])
             }
             PushPayload::DeviceConnected(DeviceConnectedPushPayload { device_name }) => {
+                self.attached_clients_cache = None;
```

Comment:
👍 ; we'll probably find we reach a point where want to abstract this away a little more, so that the "update X state on Y push notification" logic lives a little closer to the rest of the code that manages X state, but it's tractable doing it directly here for now.